File: AutoSync/3066-minimum-operations-to-exceed-threshold-value-ii/solution.java
class Solution {
    public int minOperations(int[] nums, int k) {
        return  Method(nums,k);
    }
    public static int Method(int [] nums,int k){
        PriorityQueue<Long> queue = new PriorityQueue<>();
        for (long num : nums) {
            queue.offer(num);
        }
        int count = 0;
        while(queue.peek() < k){
            long x = queue.poll(),y = queue.poll();
            queue.offer(x*2+y);
            count ++;
        }
        return count;
    }
}

File: AutoSync/3066-minimum-operations-to-exceed-threshold-value-ii/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int count = 0 ;
	priority_queue<long long,vector<long long>,greater<long long>> queue(nums.begin(),nums.end()); 
	while(queue.top() < k){
		long long x = queue.top();queue.pop();
		long long y = queue.top();queue.pop();
		queue.push(x*2+y);
		count++;
	}
	return count;
    }
};
